diff --git a/node_modules/@d4kmor/search/src/RocketSearchCombobox.js b/node_modules/@d4kmor/search/src/RocketSearchCombobox.js
index 6db88a1..f66e73b 100644
--- a/node_modules/@d4kmor/search/src/RocketSearchCombobox.js
+++ b/node_modules/@d4kmor/search/src/RocketSearchCombobox.js
@@ -2,6 +2,15 @@ import { css } from '@lion/core';
 import { LionCombobox } from '@lion/combobox';
 import { withDropdownConfig } from '@lion/overlays';
 
+const tmpl = document.createElement('template');
+tmpl.innerHTML = `
+  <button arrow-left>
+    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 40">
+      <path d="M34 20a.872.872 0 00-.428-.751L20.31 6.25a.875.875 0 00-1.226 1.25l11.861 11.625H6.875a.876.876 0 000 1.75H31.02L19.158 32.5a.877.877 0 00.613 1.5.87.87 0 00.612-.25l13.353-13.089a.876.876 0 00.264-.626l-.001-.017L34 20z" />
+    </svg>
+  </button>
+`;
+
 export class RocketSearchCombobox extends LionCombobox {
   static get styles() {
     return [
@@ -80,7 +89,7 @@ export class RocketSearchCombobox extends LionCombobox {
 
         :host([show-input]) .input-group {
           height: 70px;
-          border-bottom: 1px solid #ccc;
+          border-bottom: 1px solid var(--rocket-search-input-overlay-border-color, #ccc);
         }
 
         :host([show-input]) .input-group__prefix {
@@ -114,7 +123,7 @@ export class RocketSearchCombobox extends LionCombobox {
             position: relative;
             background: var(--rocket-search-background-color, #fff);
             display: flex;
-            border: 1px solid #dfe1e5;
+            border: 1px solid var(--rocket-search-input-border-color, #dfe1e5);
             box-shadow: none;
             border-radius: 24px;
             padding: 5px 0;
@@ -209,14 +218,6 @@ export class RocketSearchCombobox extends LionCombobox {
     return {
       ...super.slots,
       prefix: () => {
-        const tmpl = document.createElement('template');
-        tmpl.innerHTML = `
-          <button arrow-left>
-            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 40">
-              <path d="M34 20a.872.872 0 00-.428-.751L20.31 6.25a.875.875 0 00-1.226 1.25l11.861 11.625H6.875a.876.876 0 000 1.75H31.02L19.158 32.5a.877.877 0 00.613 1.5.87.87 0 00.612-.25l13.353-13.089a.876.876 0 00.264-.626l-.001-.017L34 20z" />
-            </svg>
-          </button>
-        `;
         const arrowLeft = document.importNode(tmpl.content, true).children[0];
         arrowLeft.addEventListener('click', () => {
           if (window.innerWidth < 1024) {
