diff --git a/node_modules/@d4kmor/cli/src/RocketBuild.js b/node_modules/@d4kmor/cli/src/RocketBuild.js
index ceaf716..fedd827 100755
--- a/node_modules/@d4kmor/cli/src/RocketBuild.js
+++ b/node_modules/@d4kmor/cli/src/RocketBuild.js
@@ -31,6 +31,7 @@ async function productionBuild(config) {
     config.build && config.build.serviceWorkerFileName
       ? config.build.serviceWorkerFileName
       : 'service-worker.js';
+  const { rootDir = '.' } = config.devServer;
   const mpaConfig = createMpaConfig({
     outputDir: config.build.outputDir,
     legacyBuild: false,
@@ -39,7 +40,7 @@ async function productionBuild(config) {
       swDest: path.join(config.build.outputDir, serviceWorkerFileName),
     },
     html: {
-      rootDir: path.join(config.devServer.rootDir, '_site-dev'),
+      rootDir: path.join(rootDir, '_site-dev'),
       input: '**/*.html',
       absoluteBaseUrl: config.build.absoluteBaseUrl,
     },
@@ -48,11 +49,12 @@ async function productionBuild(config) {
   mpaConfig.plugins.push(
     copy({
       patterns: ['!(*.md|*.html)*', '_merged_assets/_static/**/*.{png,gif,jpg,json,css,svg,ico}'],
-      rootDir: path.join(config.devServer.rootDir, config.pathPrefix),
+      rootDir: path.join(rootDir, config.pathPrefix),
     }),
   );
 
-  await buildAndWrite(mpaConfig);
+  const modify = config.build.rollup ?? (x => x);
+  await buildAndWrite(modify(mpaConfig));
 }
 
 export class RocketBuild {
diff --git a/node_modules/@d4kmor/cli/src/shared/eleventyConfigEmpty.js b/node_modules/@d4kmor/cli/src/shared/eleventyConfigEmpty.js
index 7b9d113..4abb6dd 100644
--- a/node_modules/@d4kmor/cli/src/shared/eleventyConfigEmpty.js
+++ b/node_modules/@d4kmor/cli/src/shared/eleventyConfigEmpty.js
@@ -46,4 +46,5 @@ export const eleventyConfigEmpty = {
   setQuietMode: () => {},
   addExtension: () => {},
   addDataExtension: () => {},
+  on: () => {},
 };
