diff --git a/node_modules/@rocket/cli/src/RocketLint.js b/node_modules/@rocket/cli/src/RocketLint.js
index 9883cfb..3769f36 100755
--- a/node_modules/@rocket/cli/src/RocketLint.js
+++ b/node_modules/@rocket/cli/src/RocketLint.js
@@ -57,7 +57,7 @@ export class RocketLint {
 
     const { errors, message } = await checkLinks.run();
     if (errors.length > 0) {
-      if (this.config.command === 'start') {
+      if (this.config.command === 'start' || process.env.CI) {
         console.log(message);
       } else {
         throw new Error(message);
