diff --git a/node_modules/@rocket/cli/src/RocketCli.js b/node_modules/@rocket/cli/src/RocketCli.js
index 19a0fa9..2dc55ca 100644
--- a/node_modules/@rocket/cli/src/RocketCli.js
+++ b/node_modules/@rocket/cli/src/RocketCli.js
@@ -92,6 +92,10 @@ export class RocketCli {
         type: String,
         description: 'Location of Rocket configuration',
       },
+      {
+        name: 'incremental',
+        type: Boolean,
+      }
     ];
     const options = commandLineArgs(mainDefinitions, {
       stopAtFirstUnknown: true,
@@ -100,8 +104,10 @@ export class RocketCli {
     this.subArgv = options._unknown || [];
     this.argvConfig = {
       command: options.command,
+      incremental: options.incremental,
       configFile: options['config-file'],
     };
+
     this.__isSetup = false;
   }
 
@@ -114,12 +120,16 @@ export class RocketCli {
       await this.mergePresets();
 
       const elev = new RocketEleventy(_inputDirCwdRelative, outputDevDir, this);
+
+      elev.setIncrementalBuild(this.argvConfig.incremental);
+
       // 11ty always wants a relative path to cwd - why?
       const rel = path.relative(process.cwd(), path.join(__dirname));
       const relCwdPathToConfig = path.join(rel, 'shared', '.eleventy.cjs');
 
       const config = new TemplateConfig(null, relCwdPathToConfig);
       elev.config = config.getConfig();
+
       /** @type {*} */
       (elev).resetConfig();
       elev.setConfigPathOverride(relCwdPathToConfig);
diff --git a/node_modules/@rocket/cli/src/getPuppeteerBrowser.js b/node_modules/@rocket/cli/src/getPuppeteerBrowser.js
new file mode 100644
index 0000000..d37b81e
--- /dev/null
+++ b/node_modules/@rocket/cli/src/getPuppeteerBrowser.js
@@ -0,0 +1,8 @@
+import puppeteer from 'puppeteer';
+
+export async function getPuppeteerBrowser() {
+  return puppeteer.launch({
+    headless: true,
+    args: ['--no-sandbox', '--font-render-hinting=none']
+  });
+}
diff --git a/node_modules/@rocket/cli/src/normalizeConfig.js b/node_modules/@rocket/cli/src/normalizeConfig.js
index 65b5d52..5344921 100644
--- a/node_modules/@rocket/cli/src/normalizeConfig.js
+++ b/node_modules/@rocket/cli/src/normalizeConfig.js
@@ -15,6 +15,7 @@ import { readConfig } from '@web/config-loader';
 import { RocketStart } from './RocketStart.js';
 import { RocketBuild } from './RocketBuild.js';
 import { RocketLint } from './RocketLint.js';
+import { getPuppeteerBrowser } from './getPuppeteerBrowser.js';
 
 import { fileURLToPath } from 'url';
 
@@ -51,6 +52,7 @@ export async function normalizeConfig(inConfig) {
     setupEleventyComputedConfig: [],
     setupCliPlugins: [],
     eleventy: () => {},
+    incremental: false,
     command: 'help',
     watch: true,
     createSocialMediaImages: true,
@@ -63,6 +65,8 @@ export async function normalizeConfig(inConfig) {
 
     ...inConfig,
 
+    puppeteerBrowser: inConfig.puppeteerBrowser ?? await getPuppeteerBrowser(),
+
     /** @type{RocketCliOptions['before11ty'][]} */
     __before11tyFunctions: [],
 
diff --git a/node_modules/@rocket/cli/src/shared/.eleventy.cjs b/node_modules/@rocket/cli/src/shared/.eleventy.cjs
index c215e6d..01fdeec 100644
--- a/node_modules/@rocket/cli/src/shared/.eleventy.cjs
+++ b/node_modules/@rocket/cli/src/shared/.eleventy.cjs
@@ -19,6 +19,10 @@ const defaultSetupUnifiedPlugins = [
 module.exports = function (eleventyConfig) {
   const config = getComputedConfig();
 
+  eleventyConfig.on('afterBuild', async () => {
+    await config.puppeteerBrowser?.close?.();
+  });
+
   const { pathPrefix, _inputDirCwdRelative, outputDevDir } = config;
 
   let metaPlugins = [
