diff --git a/node_modules/@rocket/cli/src/RocketBuild.js b/node_modules/@rocket/cli/src/RocketBuild.js
index b42ce7c..7f7e181 100755
--- a/node_modules/@rocket/cli/src/RocketBuild.js
+++ b/node_modules/@rocket/cli/src/RocketBuild.js
@@ -29,7 +29,7 @@ async function productionBuild(config) {
       name: 'copy',
       plugin: copy,
       options: {
-        patterns: ['!(*.md|*.html)*', '_merged_assets/_static/**/*.{png,gif,jpg,json,css,svg,ico}'],
+        patterns: ['!(*.md|*.html)*', '_merged_assets/_static/**/*'],
         rootDir: config.outputDevDir,
       },
     }),
diff --git a/node_modules/@rocket/cli/src/RocketCli.js b/node_modules/@rocket/cli/src/RocketCli.js
index a7f8d44..5fff5cc 100644
--- a/node_modules/@rocket/cli/src/RocketCli.js
+++ b/node_modules/@rocket/cli/src/RocketCli.js
@@ -31,6 +31,8 @@ export class RocketEleventy extends Eleventy {
 
   async write() {
     await this.__rocketCli.mergePresets();
+    for (const fn of this.__rocketCli.config.__before11tyFunctions)
+      await fn();
     await super.write();
     await this.__rocketCli.update();
   }
diff --git a/node_modules/@rocket/cli/src/eleventy-plugins/rocketCopy.cjs b/node_modules/@rocket/cli/src/eleventy-plugins/rocketCopy.cjs
index f062d16..4eb2932 100644
--- a/node_modules/@rocket/cli/src/eleventy-plugins/rocketCopy.cjs
+++ b/node_modules/@rocket/cli/src/eleventy-plugins/rocketCopy.cjs
@@ -1,6 +1,6 @@
 const rocketCopy = {
-  configFunction: (eleventyConfig, { _inputDirCwdRelative, filesExtensionsToCopy }) => {
-    eleventyConfig.addPassthroughCopy(`${_inputDirCwdRelative}/**/*.{${filesExtensionsToCopy}}`);
+  configFunction: (eleventyConfig, { _inputDirCwdRelative }) => {
+    eleventyConfig.addPassthroughCopy(`${_inputDirCwdRelative}/**/*`);
   },
 };
 
diff --git a/node_modules/@rocket/cli/src/normalizeConfig.js b/node_modules/@rocket/cli/src/normalizeConfig.js
index a4d7ff8..75182b2 100644
--- a/node_modules/@rocket/cli/src/normalizeConfig.js
+++ b/node_modules/@rocket/cli/src/normalizeConfig.js
@@ -61,6 +61,9 @@ export async function normalizeConfig(inConfig) {
     devServer: {},
 
     ...inConfig,
+
+    __before11tyFunctions: [],
+
     imagePresets: {
       responsive: {
         widths: [600, 900, 1640],
@@ -159,6 +162,9 @@ export async function normalizeConfig(inConfig) {
     if (preset.setupCliPlugins) {
       config.setupCliPlugins = [...config.setupCliPlugins, ...preset.setupCliPlugins];
     }
+    if (typeof preset.before11ty === 'function') {
+      config.__before11tyFunctions.push(preset.before11ty);
+    }
   }
   // add "local" preset
   config._presetPathes.push(path.resolve(_inputDirCwdRelative));
@@ -186,6 +192,10 @@ export async function normalizeConfig(inConfig) {
     plugins.push(pluginInst);
   }
 
+  if (typeof config.before11ty === 'function') {
+    config.__before11tyFunctions.push(config.before11ty)
+  }
+
   // TODO: check pathPrefix to NOT have a '/' at the end as it will mean it may get ignored by 11ty 🤷‍♂️
 
   return {
diff --git a/node_modules/@rocket/cli/src/shared/.eleventy.cjs b/node_modules/@rocket/cli/src/shared/.eleventy.cjs
index 881aa10..4eaa98c 100644
--- a/node_modules/@rocket/cli/src/shared/.eleventy.cjs
+++ b/node_modules/@rocket/cli/src/shared/.eleventy.cjs
@@ -30,10 +30,7 @@ module.exports = function (eleventyConfig) {
     {
       name: 'rocket-copy',
       plugin: rocketCopy,
-      options: {
-        _inputDirCwdRelative,
-        filesExtensionsToCopy: 'png,gif,jpg,jpeg,svg,css,xml,json,js',
-      },
+      options: { _inputDirCwdRelative },
     },
     {
       name: 'eleventy-plugin-mdjs-unified',
