diff --git a/node_modules/@rocket/cli/src/RocketLint.js b/node_modules/@rocket/cli/src/RocketLint.js
index 9883cfb..121c303 100755
--- a/node_modules/@rocket/cli/src/RocketLint.js
+++ b/node_modules/@rocket/cli/src/RocketLint.js
@@ -49,15 +49,27 @@ export class RocketLint {
     }
 
     const checkLinks = new CheckHtmlLinksCli();
-    checkLinks.setOptions({
-      rootDir: this.config.lintInputDir,
-      printOnError: false,
-      continueOnError: true,
-    });
+checkLinks.setOptions({
+  rootDir: this.config.lintInputDir,
+  printOnError: false,
+  continueOnError: true,
+  ignoreLinkPatterns: [
+    '*/_assets/_static/docs-playground.js',
+    '*_assets/_static/docs-playground.js',
+    '../_assets/_static/docs-playground.js',
+    '../../_assets/_static/docs-playground.js',
+    '../../../_assets/_static/docs-playground.js',
+    '../../../../_assets/_static/docs-playground.js',
+    '../../../../../_assets/_static/docs-playground.js',
+    '../../../../../../_assets/_static/docs-playground.js',
+    '../../../../../../../_assets/_static/docs-playground.js',
+  ],
+});
 
     const { errors, message } = await checkLinks.run();
+
     if (errors.length > 0) {
-      if (this.config.command === 'start') {
+      if (this.config.command === 'start' || process.env.CI) {
         console.log(message);
       } else {
         throw new Error(message);
diff --git a/node_modules/@rocket/cli/src/eleventy-plugins/insertResponsiveImages.cjs b/node_modules/@rocket/cli/src/eleventy-plugins/insertResponsiveImages.cjs
index 349831f..4a2f4e8 100644
--- a/node_modules/@rocket/cli/src/eleventy-plugins/insertResponsiveImages.cjs
+++ b/node_modules/@rocket/cli/src/eleventy-plugins/insertResponsiveImages.cjs
@@ -99,7 +99,7 @@ function getImages(html) {
         const src = getAttribute(data, 'src');
         const title = getAttribute(data, 'title');
         const alt = getAttribute(data, 'alt');
-        if (presetName) {
+        if (presetName && !src.match(/^https?\:\/\//)) {
           images.push({
             presetName,
             attributes,
