diff --git a/node_modules/@rocket/eleventy-rocket-nav/eleventy-rocket-nav.js b/node_modules/@rocket/eleventy-rocket-nav/eleventy-rocket-nav.js
index c81108b..dea9709 100644
--- a/node_modules/@rocket/eleventy-rocket-nav/eleventy-rocket-nav.js
+++ b/node_modules/@rocket/eleventy-rocket-nav/eleventy-rocket-nav.js
@@ -68,7 +68,7 @@ const headingsCache = new Map();
  * @param {NavigationNode[]} nodes
  * @param {string} key
  */
-function findNavigationEntries(nodes = [], key = '') {
+function findNavigationEntries(nodes = [], key = '', { additionalHeadings = [] } = {}) {
   const pages = [];
   for (const entry of nodes) {
     if (entry.data && entry.data.eleventyNavigation) {
@@ -101,7 +101,7 @@ function findNavigationEntries(nodes = [], key = '') {
           );
         }
         const headings = /** @type {Heading[]} */ (headingsCache.get(entry.templateContent.html));
-        const anchors = headings.map(heading => ({
+        const anchors = [...headings, ...additionalHeadings.map(text => ({ text }))].map(heading => ({
           key: heading.text + Math.random(),
           parent: entry.key,
           url: `${entry.url}#${heading.id}`,
