diff --git a/node_modules/@rocket/eleventy-rocket-nav/eleventy-rocket-nav.js b/node_modules/@rocket/eleventy-rocket-nav/eleventy-rocket-nav.js
index 5abdc21..c039cb0 100644
--- a/node_modules/@rocket/eleventy-rocket-nav/eleventy-rocket-nav.js
+++ b/node_modules/@rocket/eleventy-rocket-nav/eleventy-rocket-nav.js
@@ -231,7 +231,7 @@ function navigationToHtml(pages, _options = {}) {
           output.push(
             `<${options.listItemElement}${liClass.length ? ` class="${liClass.join(' ')}"` : ''}>`,
           );
-          output.push(`<a href="${urlFilter(entry.url)}">${entry.title}</a>`);
+          output.push(`<a href="${urlFilter(entry.url, '/')}">${entry.title}</a>`);
           if (options.showExcerpt && entry.excerpt) {
             output.push(`: ${entry.excerpt}`);
           }
