diff --git a/node_modules/bind-decorator/index.js b/node_modules/bind-decorator/index.js
index 423be2d..1dd2b64 100644
--- a/node_modules/bind-decorator/index.js
+++ b/node_modules/bind-decorator/index.js
@@ -1,11 +1,10 @@
 "use strict";
-Object.defineProperty(exports, "__esModule", { value: true });
 var constants;
 (function (constants) {
     constants.typeOfFunction = 'function';
     constants.boolTrue = true;
 })(constants || (constants = {}));
-function bind(target, propertyKey, descriptor) {
+export function bind(target, propertyKey, descriptor) {
     if (!descriptor || (typeof descriptor.value !== constants.typeOfFunction)) {
         throw new TypeError("Only methods can be decorated with @bind. <" + propertyKey + "> is not a method!");
     }
@@ -23,5 +22,4 @@ function bind(target, propertyKey, descriptor) {
         }
     };
 }
-exports.bind = bind;
-exports.default = bind;
+export default bind;
