diff --git a/node_modules/rocket-preset-slide-decks/_includes/_joiningBlocks/_layoutDeck/content/10-content.njk b/node_modules/rocket-preset-slide-decks/_includes/_joiningBlocks/_layoutDeck/content/10-content.njk
index 350b5cd..f8ed737 100644
--- a/node_modules/rocket-preset-slide-decks/_includes/_joiningBlocks/_layoutDeck/content/10-content.njk
+++ b/node_modules/rocket-preset-slide-decks/_includes/_joiningBlocks/_layoutDeck/content/10-content.njk
@@ -1,12 +1 @@
 {{ content | safe }}
-
-<script async defer>
-  addEventListener('keyup', event => {
-    if (event.key !== 'f')
-      return;
-    else if (document.fullscreen)
-      document.exitFullscreen();
-    else
-      document.body.requestFullscreen();
-  });
-</script>
diff --git a/node_modules/rocket-preset-slide-decks/_includes/layout-deck.njk b/node_modules/rocket-preset-slide-decks/_includes/layout-deck.njk
index 3a74498..0f26cc4 100644
--- a/node_modules/rocket-preset-slide-decks/_includes/layout-deck.njk
+++ b/node_modules/rocket-preset-slide-decks/_includes/layout-deck.njk
@@ -28,13 +28,16 @@
                  transition="{{ transition or 'slide-in slide-out' }}"
                  itemscope
                  itemtype="http://schema.org/PresentationDigitalDocument"
+                 {{ 'unbind' if unbind }}
                  loading>
+      {% if not unbind %}
       <gluon-keybinding key="Backspace" slot="backward"></gluon-keybinding>
       <gluon-keybinding key=" " slot="forward"></gluon-keybinding>
       <gluon-keybinding key="h" slot="backward"></gluon-keybinding>
       <gluon-keybinding key="j" slot="forward"></gluon-keybinding>
       <gluon-keybinding key="k" slot="backward"></gluon-keybinding>
       <gluon-keybinding key="l" slot="forward"></gluon-keybinding>
+      {% endif %}
 
       {% for slide in collections.slides %}
         {% set isPageSlide =
@@ -58,3 +61,14 @@
   {% include 'partials/bottom.njk' %}
   </body>
 </html>
+
+<script async defer>
+  addEventListener('keyup', event => {
+    if (event.defaultPrevented || event.key !== 'f')
+      return;
+    else if (document.fullscreen)
+      document.exitFullscreen();
+    else
+      document.body.requestFullscreen();
+  });
+</script>
diff --git a/node_modules/rocket-preset-slide-decks/rocket-preset-slide-decks.js b/node_modules/rocket-preset-slide-decks/rocket-preset-slide-decks.js
index 70b81ff..6d3e680 100644
--- a/node_modules/rocket-preset-slide-decks/rocket-preset-slide-decks.js
+++ b/node_modules/rocket-preset-slide-decks/rocket-preset-slide-decks.js
@@ -6,6 +6,11 @@ import { dirname, join, resolve } from 'path';
 import { fileURLToPath } from 'url';
 import { addPlugin, adjustPluginOptions } from 'plugins-manager';
 import { slideDecksPlugin } from './eleventy/slide-decks.js';
+import { markdownDirectives } from 'rocket-preset-markdown-directives';
+import remarkDirective from 'remark-directive';
+import visit from 'unist-util-visit';
+// import { visit } from 'unist-util-visit';
+import { h } from 'hastscript';
 
 const path = resolve(dirname(fileURLToPath(import.meta.url)));
 
@@ -54,5 +59,47 @@ export function slideDecks() {
       }),
     ],
 
+    setupUnifiedPlugins: [
+      addPlugin({
+        name: 'markdown-directives',
+        plugin: markdownDirectives,
+        location: 'top',
+      }),
+      addPlugin({
+        name: 'remark-directive',
+        plugin: remarkDirective,
+        location: 'top',
+      }),
+      addPlugin({
+        name: 'slide-decks-remark-directives',
+        plugin() {
+          return tree => {
+            visit(tree, node => {
+              switch (node.type) {
+                case 'textDirective':
+                case 'leafDirective':
+                case 'containerDirective': {
+                  try {
+                  const data = node.data || (node.data = {});
+                  const hast = h(node.name, node.attributes)
+                  data.hName = hast.tagName;
+                  data.hProperties = hast.properties;
+                }catch(e) {console.error(e)}
+                }
+              }
+            })
+          }
+        },
+        location: 'remark-directive',
+        how: 'after',
+      }),
+      adjustPluginOptions(
+        'markdown-directives',
+        /** @type{import('rocket-preset-markdown-directives').Config} */
+        ({
+          reveal: () => ({ tagName: 'section', attributes: { reveal: true } }),
+        })
+      ),
+    ],
   };
 }
