diff --git a/node_modules/slidem/slidem-deck.js b/node_modules/slidem/slidem-deck.js
index 503d779..ec33af2 100644
--- a/node_modules/slidem/slidem-deck.js
+++ b/node_modules/slidem/slidem-deck.js
@@ -78,23 +78,32 @@ document.head.appendChild(styleNode);
 export class SlidemDeck extends GluonElement {
   get template() {
     return html`
-      <div class="slides">
+      <div class="slides" part="slides">
         <slot id="slides"></slot>
       </div>
-      <div id="progress"></div>
-      <div id="timer"></div>
+      <div id="progress" part="progress"></div>
+      <div id="timer" part="timer"></div>
+      ${this.hasAttribute('unbind') ? html`
+      <button hidden id="timerToggle"></button>
+      <button hidden id="presenterToggle"></button>
+      ` : html`
       <gluon-keybinding id="timerToggle" key="t"></gluon-keybinding>
       <gluon-keybinding id="presenterToggle" key="p"></gluon-keybinding>
+      `}
       <div id="forward">
+      ${this.hasAttribute('unbind') ? '' : html`
         <gluon-keybinding key="PageDown"></gluon-keybinding>
         <gluon-keybinding key="ArrowRight"></gluon-keybinding>
         <gluon-keybinding key="Right"></gluon-keybinding>
+      `}
         <slot name="forward"></slot>
       </div>
       <div id="backward">
+      ${this.hasAttribute('unbind') ? '' : html`
         <gluon-keybinding key="PageUp"></gluon-keybinding>
         <gluon-keybinding key="ArrowLeft"></gluon-keybinding>
         <gluon-keybinding key="Left"></gluon-keybinding>
+      `}
         <slot name="backward"></slot>
       </div>
       <style>
diff --git a/node_modules/slidem/slidem-slide-base.js b/node_modules/slidem/slidem-slide-base.js
index a36279d..7a4af43 100644
--- a/node_modules/slidem/slidem-slide-base.js
+++ b/node_modules/slidem/slidem-slide-base.js
@@ -7,6 +7,11 @@ const styleText = document.createTextNode(`
     opacity: 0;
     transition: opacity 0.2s;
   }
+
+  [current],
+  [past] {
+    opacity: 1;
+  }
 `);
 
 const styleNode = document.createElement('style');
@@ -20,6 +25,7 @@ const slidemStyle = html`
       flex-direction: row;
       overflow: hidden;
       align-items: center;
+      background: var(--background);
       background-size: cover;
       background-position: center;
     }
@@ -110,6 +116,9 @@ export class SlidemSlideBase extends GluonElement {
     super.connectedCallback();
     this._steps = Array.from(this.querySelectorAll('[reveal]'));
     this.steps = this._steps.length;
+    this._steps.forEach((step, i) => step.setAttribute('step', i + 2));
+    if (this._steps.length)
+      this._steps[0].previousElementSibling.setAttribute('step', 1);
     this.__resizeContent();
     let resizeTimeout;
     window.addEventListener('resize', () => {
@@ -143,13 +152,14 @@ export class SlidemSlideBase extends GluonElement {
     return Number(this.getAttribute('step')) || 1;
   }
 
-  __setStep(newStep) {
-    this._steps.forEach((step, i) => {
-      if (i < newStep - 1) {
-        step.style.opacity = 1;
-      } else {
-        step.style.opacity = 0;
-      }
+  __setStep(step) {
+    this.querySelector('[step="1"]')?.toggleAttribute?.('past', step > 1);
+    this._steps.forEach((el, i) => {
+      const elStep = i + 2;
+      const past = elStep < step;
+      const current = elStep === step;
+      el.toggleAttribute('past', past);
+      el.toggleAttribute('current', current);
     });
   }
 
diff --git a/node_modules/slidem/slidem-slide.js b/node_modules/slidem/slidem-slide.js
index 4cdc5a1..f8a80ea 100644
--- a/node_modules/slidem/slidem-slide.js
+++ b/node_modules/slidem/slidem-slide.js
@@ -28,13 +28,8 @@ export class SlidemSlide extends SlidemSlideBase {
     super.connectedCallback();
     const background = this.getAttribute('background');
     if (background) {
-      if (background.match(/^--[a-zA-Z-]*$/)) {
-        // Workaround for IE11 lacking CSS variables
-        if (window.ShadyCSS && window.ShadyCSS.variables) {
-          this.style.background = window.ShadyCSS.variables[background];
-        } else {
-          this.style.background = `var(${background})`;
-        }
+      if (background.match(/^--/)) {
+        this.style.setProperty('--background', `var(${background})`);
       } else if (background.match(/^(http|\/|\.)/)) {
         let image = `url(${background})`;
         const darken = this.getAttribute('darken-background');
@@ -43,7 +38,7 @@ export class SlidemSlide extends SlidemSlideBase {
         }
         this.style.backgroundImage = image;
       } else {
-        this.style.background = background;
+        this.style.setProperty('--background', background);
       }
     }
 
@@ -103,6 +98,8 @@ export class SlidemSlide extends SlidemSlideBase {
 
   attributeChangedCallback(attr, oldVal, newVal) {
     super.attributeChangedCallback(attr, oldVal, newVal);
+    if (attr === 'active')
+      this.dispatchEvent(new Event('activated'));
     if (attr === 'active' || attr === 'next') {
       if (newVal !== null) {
         this.__rescale();
